src/fsharp/symbols/Exprs.fs

```diff
@@ -829,7 +829,7 @@ module FSharpExprConvert =
                 let typR = ConvType cenv (mkAppTy tycr tyargs)
                 E.UnionCaseTag(ConvExpr cenv env arg1, typR) 
 
-            | TOp.TraitCall (TTrait(tys, nm, memFlags, argtys, _rty, _solution)), _, _                    -> 
+            | TOp.TraitCall (TTrait(tys, nm, memFlags, argtys, _rty, _solution, _traitCtxt)), _, _ -> 
                 let tysR = ConvTypes cenv tys
                 let tyargsR = ConvTypes cenv tyargs
                 let argtysR = ConvTypes cenv argtys
```

Comment:
@dsyme I was hoping that this or #6810 would extend the TypedTree with information that could help Fable (or other compilers) resolve the SRTPs, but I see `BasicPatterns.TraitCall` remains unchanged so it's going to be very difficult to support this feature.

Would it be possible to expose the TraitContext and/or any other information that may help with the resolution? The fifth argument of the TraitCall active pattern, btw (which corresponds here to `tyargsR`) seems to be always empty. Is it supposed to be for the return types?